x-pack/plugins/canvas/public/components/app/index.tsx

```diff
@@ -28,12 +31,21 @@ class ShortcutManagerContextWrapper extends React.Component<React.PropsWithChild
   }
 }
 
-export const App: FC<{ history: ScopedHistory }> = ({ history }) => {
-  const { updatePath } = useNavLinkService();
-
+export const App: FC<{ history: ScopedHistory; appUpdater: BehaviorSubject<AppUpdater> }> = ({
+  history,
+  appUpdater,
+}) => {
   useEffect(() => {
     return history.listen(({ pathname, search }) => {
-      updatePath(pathname + search);
+      const path = pathname + search;
+      appUpdater.next(() => ({
+        defaultPath: `${path}`,
+      }));
+
+      getSessionStorage().set(
+        `${SESSIONSTORAGE_LASTPATH}:${coreServices.http.basePath.get()}`,
+        path
+      );
```

Comment:
This is the only place the old `navLinksService.updatePath` was called - so I just deleted this service and did this inline, instead.